language/move-vm/runtime/src/native_functions.rs

```diff
@@ -166,11 +111,35 @@ impl<'a, L: LogContext> NativeContext for FunctionContext<'a, L> {
         }
     }
 
-    fn type_to_type_layout(&self, ty: &Type) -> PartialVMResult<Option<MoveTypeLayout>> {
+    pub fn type_to_type_layout(&self, ty: &Type) -> PartialVMResult<Option<MoveTypeLayout>> {
         match self.resolver.type_to_type_layout(ty) {
             Ok(ty_layout) => Ok(Some(ty_layout)),
             Err(e) if e.major_status().status_type() == StatusType::InvariantViolation => Err(e),
             Err(_) => Ok(None),
         }
     }
 }
+
+impl<'a, 'b> move_vm_types::natives::function::NativeContext for NativeContext<'a, 'b> {
```

Comment:
Right now I have to keep the trait due to values (vector ops) being in a separate crate. If we merge `move-vm-runtime` and `move-vm-values` then this trait can go away. I'll investigate this more and see what's the right thing to do here.

Reply:
Not sure if I follow. The vector ops are coming in as function pointers right? so I don't see how the separate crate is an issue (all of the native functions are a separate crate)

Reply:
Can we just move the vector ops somewhere else?

Reply:
Hmm.. It might be tricky to move the actual implementation of the vector ops out of the crate since it needs to access the internals of the values. However I noticed that the native context is only used to verify that the layout of the vector value matches the type argument, which can be moved to the natives.